src/test/java/jewellery/inventory/unit/service/ProductServiceTest.java

```diff
@@ -0,0 +1,264 @@
+    resourceQuantityRequestDto = createResourceQuantityRequestDto(pearl);
+    productRequestDto = createProductRequestDto(user, resourceQuantityRequestDto);
+  }
+
+  @Test
+  void testDeleteProductContainsProductBreakDown() {
+    testProduct.setProductsContent(List.of(new Product(), new Product(), new Product()));
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+
+    productService.deleteProduct(testProduct.getId());
+
+    assertEquals(0, testProduct.getProductsContent().size());
+  }
+
+  @Test
+  void testDeleteProductSuccessfully() {
+
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+
+    productService.deleteProduct(testProduct.getId());
+
+    assertEquals(0, productRepository.count());
+  }
+
+  @Test
+  void testDeleteProductShouldThrowExceptionWhenProductIsPartOfProduct() {
+    testProduct.setContentOf(new Product());
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+    UUID productId = testProduct.getId();
+    assertThrows(ProductIsContentException.class, () -> productService.deleteProduct(productId));
+  }
+
+  @Test
+  void testDeleteProductShouldThrowExceptionWhenProductIsSold() {
+    testProduct.setSold(true);
+    when(productRepository.findById(testProduct.getId())).thenReturn(Optional.of(testProduct));
+    UUID productId = testProduct.getId();
+    assertThrows(ProductIsSoldException.class, () -> productService.deleteProduct(productId));
+  }
+
+  @Test
+  void testDeleteProductGetProductShouldThrowExceptionWhenProductNotExist() {
+    UUID fakeId = UUID.randomUUID();
+    assertThrows(ProductNotFoundException.class, () -> productService.deleteProduct(fakeId));
+  }
+
+  @Test
+  void createProductSuccessfully() {
+
+    when(userRepository.findById(productRequestDto.getOwnerId())).thenReturn(Optional.of(user));
+    when(resourceInUserRepository.findByResourceIdAndOwnerId(pearl.getId(), user.getId()))
+        .thenReturn(Optional.of(resourceInUser));
+    user.setResourcesOwned(List.of(resourceInUser));
+
+    ProductResponseDto response = new ProductResponseDto();
+    when(productMapper.mapToProductResponseDto(any())).thenReturn(response);
+
+    ProductResponseDto actual = productService.createProduct(productRequestDto);
+
+    assertEquals(actual, response);
+  }
+
+  @Test
+  void testCreateProductShouldRemoveResourceFromUserWhenResourceInUserIsZero() {
+    when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+    when(resourceInUserRepository.findByResourceIdAndOwnerId(pearl.getId(), user.getId()))
+        .thenReturn(Optional.of(resourceInUser));
+    resourceInUser.setQuantity(5);
+    user.setResourcesOwned(List.of(resourceInUser));
+    resourceQuantityRequestDto.setQuantity(5);
+
+    productService.createProduct(productRequestDto);
+
+    assertEquals(0, resourceInUserRepository.count());
+  }
```

Comment:
This is not needed considering we are utilizing the resourceInUserService's functionality and it would be mocked here. Furthermore, resourceInUserRepository i smocked so unless we stub it, it's always going to return 0.

Reply:
Removed!